packages/app-builder-lib/src/targets/snap.ts

```diff
@@ -195,12 +195,19 @@ export default class SnapTarget extends Target {
       Icon: "${SNAP}/meta/gui/icon.png",
     })
 
+    const extraAppArgs: Array<string> = options.executableArgs ?? []
     if (this.isElectronVersionGreaterOrEqualThan("5.0.0") && !isBrowserSandboxAllowed(snap)) {
-      args.push("--extraAppArgs=--no-sandbox")
+      const noSandboxArg = "--no-sandbox"
+      if (!extraAppArgs.includes(noSandboxArg)) {
+        extraAppArgs.push(noSandboxArg)
+      }
       if (this.isUseTemplateApp) {
         args.push("--exclude", "chrome-sandbox")
       }
     }
+    if (extraAppArgs.length > 0) {
+      args.push("--extraAppArgs=" + extraAppArgs.join(" "))
```

Comment:
If `extraAppArgs = ['--test', '--test2']`, then this would fail, right?
It would be `--extraAppArgs=--test --test2`. Do we need to wrap it in double quotes? (i.e. `--extraAppArgs="--test --test2 --no-sandbox"`)

Reply:
I would simplify this to be:
```
const executableArgs: string[] = options.executableArgs ?? []
if (this.isElectronVersionGreaterOrEqualThan("5.0.0") && !isBrowserSandboxAllowed(snap)) {
  executableArgs.push('--no-sandbox')
  if (this.isUseTemplateApp) {
    args.push("--exclude", "chrome-sandbox")
  } 
}
const extraArgs = new Array(new Set(executableArgs)) // remove all duplicate entries
if (extraArgs.length) {
  args.push(`--extraAppArgs="${extraArgs.join(" ")}"`)  
}
```

Reply:
I tested it with 

```
executableArgs: ["--no-sandbox", "--disable-seccomp-filter-sandbox"]
```

for solving a problem of my app similar to [this one here](https://github.com/snapcrafters/mattermost-desktop/pull/46) and it's working as expected for my app. The args are passed to underlaying os execv-like system call as string array as the argument instead of a piece of shell argument string, the `app-builder-bin` shall receive the value as is, so we probably don't need to escape the value there

for removing duplicate arguments, personally I would prefer to lean toward more of keeping the argument as is, because

- the order of argument might matter, like `["--run-a", "--run-b"]` and `["--run-b", "--run-a"]` might have different implication 
- repeating argument might be meaningful for some cases, such as `["--verbose", "--verbose", "--verbose"]`, while this might not be the best example and best argument design in the world, there might still be some legit use cases where you want to pass repeating argument, like you want to test your electron argument handler and ensure when seeing duplicate argument it throws error

Reply:
Very interesting and great reasoning. I like it 👍

Reply:
Quick note: Since this could impact all linux builds with `executableArgs` set and this was never supported for Snap, I would consider this more of a feature or breaking change. (Luckily we have v24-alpha en route so we can include this there as a Breaking Change just to play it safe)